google/codelab/first_codelab_boot.sh

```diff
@@ -14,15 +14,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-PYTHONPATH=/opt/spinnaker/pylib python /opt/spinnaker/install/codelab_config.py
-echo "debianRepository: http://$(hostname):9999/ trusty main" \
-    > /opt/spinnaker/config/rosco-local.yml
-chown spinnaker:spinnaker /opt/spinnaker/config/rosco-local.yml
+hal config ci jenkins enable
+echo admin | hal config ci jenkins master add CodelabJenkins --address http://localhost:5656 --username admin --password
```

Comment:
I'm not sure about this. Might it be possible to use a unique password? The pattern in GCE is (or at least used to be) that passwords are written to the VM metadata as a label. I actually dont like that practice, but that seems to be it and is better than a hardcoded default. Since you are configuring hal here with the password, you can also give hal that password so spinnaker knows. The user would just need to check for the password in the metadata. Or to the file where the password is. Jenkins has a file where it puts the password. I forget what that is offhand. You could do that (which I'd recommend for starters). IN summary:
   1) Put a unique password in the place jenkins wants it if not already there.
       be sure it is chmod 400 and owned by the jenkins user.
   2) feed the contents of that file to hal
   3) Document a reminder to the user where they can learn what the password is. They will need root access.

Reply:
@ewiseblatt : Is the suggestion here that we generate a unique password each time that a user provisions an instance based on the image, or that we simply pick something better than 'admin' but that will be the same each time an instance is provisioned?  Given that is is a codelab image and not intended for production use, I want to make sure that we're not making it overly complex for a user to follow the instructions.

Reply:
I looked a bit more into options here, and it feels like this might be more effort that it's worth (though certainly not impossible).  (It's also worth noting that the admin/admin username/password is pre-existing in the codelab, I've just changed the code to set it up to use Halyard.)

On initial install Jenkins can generate a random password, but as currently set up we install Jenkins before we take an image, so it would be the same random password for all installs.

We could ask the user to supply a Jenkins password in instance metadata, or just generate a random one when spinning up the codelab image; there does not seem to be an easy way to change the password on a Jenkins instance via script (beyond generated the salted hash of the chosen password and inserting it into the Jenkins user's XML file).

I'm leaning towards deferring making this change in the current PR, which was intended just to unbreak the codelab (which is currently broken) but not make any further functional changes---but happy to discuss further.

Reply:
Isnt the first_boot script run when the instance is first booted? That's where you'd generate a password and tell halyard about it.

Reply:
`first_codelab_boot.sh` is run when the instance is started, but by that time Jenkins has already been installed and configured with a password (before we generated the image).  So we'd need to splice a new password's hash into the existing Jenkins config file.